modules/seqexec/server/src/main/scala/seqexec/server/gmos/GmosHeader.scala

```diff
@@ -96,21 +122,23 @@ object GmosHeader {
             buildString(gmosReader.dcName, KeywordName.GMOSDC),
             buildString(gmosReader.detectorType, KeywordName.DETTYPE),
             buildString(gmosReader.detectorId, KeywordName.DETID),
-            buildDouble(gmosReader.exposureTime, KeywordName.EXPOSURE),
+            buildDouble(
+              gmosObsReader.isNS.ifM(
+                (gmosObsReader.nodCount, gmosReader.exposureTime).mapN{ case (c, t) => 2 * c * t },
```

Comment:
I tend to put these calculations on the keyword reader and leave the `Header` class just as set of values and keyword names. WDYT?

Reply:
I agree in general, but the problem here is that the calculation combines values from two readers.

Reply:
Good point